src/Vacuum/Vacuum_math.jl

```diff
@@ -0,0 +1,728 @@
+- `dx1::Float64`           : 각 인덱스에 추가되는 미세 오프셋 (기본값 0, `ai = (i-1) + dx1`)
+
+# 반환값
+- `vecout::Vector{Float64}` : 변환된 출력 배열 (길이 `mth`)
+"""
+function trans(vecin::Vector{Float64}, mth::Int; dx0=0.0, dx1=0.0)
+    mthin = length(vecin)
+    ext = [vecin; vecin[1:2]]
+    x_in = range(0, 1, length=mthin+2)
+    itp = CubicSplineInterpolation(x_in, ext, extrapolation_bc=Periodic())
+    vecout = zeros(mth)
+    for i in 1:mth
+        ai = (i-1) + dx1
+        x = ai / mth + dx0 / mthin
+        x = x % 1.0  # This is for periodicity.
+        vecout[i] = itp(x)
+    end
+    return vecout
+end
+
+#############################################################
+# Searching index , replacing search, serachx
+#############################################################
+"""
+    search(xbar, x::AbstractVector{<:Real})
+
+정렬된 배열 x에서 xbar가 속하는 구간의 인덱스(1-based)를 반환.
+- xbar < x[1]이면 0 반환
+- xbar ≥ x[end]이면 length(x)-1 반환
+- 그 외: x[i] ≤ xbar < x[i+1]를 만족하는 i 반환
+"""
+function search(xbar, x::AbstractVector{<:Real})
+    n = length(x)
+    idx = searchsortedfirst(x, xbar)
+    if xbar < x[1]
+        return 0
+    elseif xbar >= x[end]
+        return n-1
+    else
+        return idx - 1
+    end
+end
+
+#############################################################
+# Legendre function of the first kind eq.(47)~(50) , replacing aleg
+#############################################################
+
+# Chance eq.(49)
+function P0_minus_half(s)
+    m1 = 2 / (s + 1)
+    return 2 / π * sqrt(m1) * ellipk(m1)
+end
+
+# Chance eq.(49) (wrong/original)
+function P0_minus_half_wrong(s)
+    m1 = 2 / (s + 1)
+    return 2 / π * sqrt(m1) * ellipkwrong(m1)
+end
+
+# Chance eq.(50)
+function P0_plus_half(s)
+    m1 = 1 / (s + sqrt(s^2 - 1))^2
+    return 2 / π * sqrt(m1) * ellipe(m1)
+end
+
+# Chance eq.(50) (wrong/original)
+function P0_plus_half_wrong(s)
+    m1 = (s + sqrt(s^2 - 1))
+    return 2 / π * sqrt(m1) * ellipewrong(1/m1^2)
+end
+
+# Chance eq.(48)
+function P1_minus_half(s)
+    return 0.5 / ((s^2 - 1)^0.5) * (P0_plus_half(s) - s * P0_minus_half(s))
+end
+
+# Chance eq.(48) (wrong/original)
+function P1_minus_half_wrong(s)
+    return 0.5 / ((s^2 - 1)^0.5) * (P0_plus_half_wrong(s) - s * P0_minus_half_wrong(s))
+end
+
+# Polynomial elliptical integrals from original Chance code
+function ellipewrong(m1)
+    # This is a placeholder for the polynomial ellipe used by chance.
+    # Not yet implemented, error.
+    error("ellipewrong is not implemented yet.")
+end
+
+function ellipkwrong(m1)
+    # This is a placeholder for the polynomial ellipk used by chance.
+    # Not yet implemented, error.
+    error("ellipkwrong is not implemented yet.")
+end
+
+"""
+    Pn_minus_half(s, n)
+
+Chance 논문 식 (47)~(50)
+calculate Pⁿ_{-1/2}(s) with recursive
+
+# Arguments
+- `s::Real` : Legendre function factor (s > 1)
+- `n::Int` : maxinum order of n (0 이상)
+
+# Returns
+- `P[end]` :  P_{-1/2}^{n}(s) value in nmax 
+
+"""
+function Pn_minus_half(s::Real, n::Int)
```

Comment:
I'd probably try to find Julia implementations of Legendre functions that would work. I suspect you'd get a performance increase over coding it yourself.